optimizely/odp/zaius_rest_api_manager.py

```diff
@@ -55,34 +55,40 @@ def sendOdpEvents(self, api_key: str, api_host: str, events: List[OdpEvent]) ->
         Returns:
             retry is True - if network or server error (5xx), otherwise False
         """
-        can_retry: bool = True
+        should_retry: bool = False
         url = f'{api_host}/v3/events'
         request_headers = {'content-type': 'application/json', 'x-api-key': api_key}
 
+        try:
+            payload_dict = json.dumps(events)
+        except TypeError as err:
+            self.logger.error(Errors.ODP_EVENT_FAILED.format(err))
+            return should_retry
+
         try:
             response = requests.post(url=url,
                                      headers=request_headers,
-                                     data=json.dumps(events),
+                                     data=payload_dict,
                                      timeout=OdpRestApiConfig.REQUEST_TIMEOUT)
 
             response.raise_for_status()
-            can_retry = False
 
         except (ConnectionError, Timeout):
             self.logger.error(Errors.ODP_EVENT_FAILED.format('network error'))
-            # we do retry, can_retry = True
-        except JSONDecodeError:
-            self.logger.error(Errors.ODP_EVENT_FAILED.format('JSON decode error'))
-            can_retry = False
-        except InvalidURL:
-            self.logger.error(Errors.ODP_EVENT_FAILED.format('invalid URL'))
-            can_retry = False
+            # retry on network errors
+            should_retry = True
         except RequestException as err:
```

Comment:
I think so, but can you confirm that this catches all other exceptions?

Reply:
@jaeopt yes, confirm. We use the same exception handling in event_dispatcher and fetch_datafile:
https://github.com/optimizely/python-sdk/blob/master/optimizely/event_dispatcher.py#L56 
https://github.com/optimizely/python-sdk/blob/master/optimizely/config_manager.py#L367

It'll catch all these: https://requests.readthedocs.io/en/latest/_modules/requests/exceptions/